samples/extensions/pipeline_binary/pipeline_binary.cpp

```diff
@@ -0,0 +1,273 @@
+
+		VK_CHECK(vkEndCommandBuffer(cmd));
+
+		// Wait at TOP_OF_PIPE so the command buffer executes only after the acquire semaphore is signaled.
+		VkPipelineStageFlags wait_stage = VK_PIPELINE_STAGE_TOP_OF_PIPE_BIT;
+		VkSubmitInfo         submit_info{};
+		submit_info.sType                = VK_STRUCTURE_TYPE_SUBMIT_INFO;
+		submit_info.waitSemaphoreCount   = 1;
+		submit_info.pWaitSemaphores      = &semaphores.acquired_image_ready;
+		submit_info.pWaitDstStageMask    = &wait_stage;
+		submit_info.commandBufferCount   = 1;
+		submit_info.pCommandBuffers      = &cmd;
+		submit_info.signalSemaphoreCount = 1;
+		submit_info.pSignalSemaphores    = &semaphores.render_complete;
+
+		// Use the graphics queue for this lightweight submission.
+		auto graphics_queue = get_device().get_queue_by_flags(VK_QUEUE_GRAPHICS_BIT, 0).get_handle();
+		VK_CHECK(vkQueueSubmit(graphics_queue, 1, &submit_info, VK_NULL_HANDLE));
+	}
+
+	submit_frame();
+}
+
+void PipelineBinary::create_compute_pipeline()
+{
+	// Empty pipeline layout (no descriptors / push constants)
+	VkPipelineLayoutCreateInfo pipeline_layout_create_info{VK_STRUCTURE_TYPE_PIPELINE_LAYOUT_CREATE_INFO};
+	VK_CHECK(vkCreatePipelineLayout(get_device().get_handle(), &pipeline_layout_create_info, nullptr, &pipeline_layout));
+
+	// Load and cache the compute shader stage once for reuse
+	compute_shader_stage = load_shader("pipeline_binary/glsl/binary_demo.comp.spv", VK_SHADER_STAGE_COMPUTE_BIT);
+	compute_shader       = compute_shader_stage.module;
+
+	// Cache the compute pipeline create info for reuse by the pipeline binary demo
+	compute_ci_cache        = {VK_STRUCTURE_TYPE_COMPUTE_PIPELINE_CREATE_INFO};
+	compute_ci_cache.stage  = compute_shader_stage;
+	compute_ci_cache.layout = pipeline_layout;
+
+	// Ensure we have a pipeline cache, as some drivers may expect one
+	if (pipeline_cache == VK_NULL_HANDLE)
+	{
+		create_pipeline_cache();
+	}
+
+	VK_CHECK(vkCreateComputePipelines(get_device().get_handle(), pipeline_cache, 1, &compute_ci_cache, nullptr, &compute_pipeline));
+}
+
+void PipelineBinary::log_pipeline_binary_support()
+{
+	VkPhysicalDevicePipelineBinaryFeaturesKHR   features{VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_PIPELINE_BINARY_FEATURES_KHR};
+	VkPhysicalDevicePipelineBinaryPropertiesKHR props{VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_PIPELINE_BINARY_PROPERTIES_KHR};
+
+	VkPhysicalDeviceFeatures2 features2{VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_FEATURES_2};
+	features2.pNext = &features;
+
+	VkPhysicalDeviceProperties2 props2{VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_PROPERTIES_2};
+	props2.pNext = &props;
+
+	vkGetPhysicalDeviceFeatures2(get_device().get_gpu().get_handle(), &features2);
+	vkGetPhysicalDeviceProperties2(get_device().get_gpu().get_handle(), &props2);
+
+	LOGI("VK_KHR_pipeline_binary supported feature: pipelineBinaries = {}", features.pipelineBinaries ? "true" : "false");
```

Comment:
This information should be display in the applications GUI instead of simply outputting it to the terminal/log.